invenio_administration/assets/semantic-ui/js/invenio_administration/details/AdminDetailsView.js

```diff
@@ -0,0 +1,173 @@
+   * Returns an array of objects, each object being composed as:
+   * {
+   *   key: str,
+   *   text: str,
+   *   order: number
+   * }
+   *
+   * @param {Object[]} columns
+   * @returns {Object[]} an array of objects, ordered by the provided order.
+   */
+  sortColumns = (columns) => {
+    let sorted = [];
+    Object.keys(columns).forEach((key) => {
+      let innerObj = columns[key];
+      if (
+        Object.prototype.hasOwnProperty.call(innerObj, "text") &&
+        Object.prototype.hasOwnProperty.call(innerObj, "order")
+      ) {
+        sorted.push({
+          key: key,
+          text: columns[key].text,
+          order: innerObj.order,
+        });
+      }
+    });
+    return sorted.sort((a, b) => a.order > b.order);
+  };
+
+  childrenWithData = (data, columns) => {
+    const { children } = this.props;
+    return React.Children.map(children, (child) => {
+      if (React.isValidElement(child)) {
+        return React.cloneElement(child, { data: data, columns: columns });
+      }
+      return child;
+    });
+  };
+
+  handleDelete = () => {
+    // after deleting the resource go back to the list view
+    const { listUIEndpoint } = this.props;
+    window.location.href = listUIEndpoint;
+  };
+
+  render() {
+    const {
+      title,
+      columns,
+      actions,
+      displayEdit,
+      displayDelete,
+      apiEndpoint,
+      idKeyPath,
+      listUIEndpoint,
+      resourceName,
+    } = this.props;
+    const { loading, data, error } = this.state;
+
+    const sortedColumns = this.sortColumns(columns);
+    return (
+      <ErrorPage
+        error={!_isEmpty(error)}
+        errorCode={error?.response.status}
+        errorMessage={error?.response.data}
+      >
+        <Grid stackable>
+          <Grid.Row columns="2">
+            <Grid.Column verticalAlign="middle">
+              <Header as="h1">{title}</Header>
+            </Grid.Column>
+            <Grid.Column
+              verticalAlign="middle"
+              floated="right"
+              textAlign="right"
+            >
+              <Actions
+                title={title}
+                resourceName={resourceName}
+                apiEndpoint={apiEndpoint}
+                displayEdit={displayEdit}
+                displayDelete={displayDelete}
+                actions={actions}
+                resource={data}
+                idKeyPath={idKeyPath}
+                successCallback={this.handleDelete}
+                listUIEndpoint={listUIEndpoint}
+              />
+            </Grid.Column>
+          </Grid.Row>
+          <Grid.Column width="16">
+            {loading ? (
+              <Segment className="rel-pt-4 rel-pb-4">
+                <Dimmer active inverted>
+                  <Loader />
+                </Dimmer>
+              </Segment>
+            ) : (
+              <>
+                <DetailsTable data={data} columns={sortedColumns} />
+                {this.childrenWithData(data, columns)}
```

Comment:
Question: I am not sure I understand this, the first one is creating the details table which can be overridden but not the sure what the second line is doing `{this.childrenWithData(data, columns)}`? As I understand it will display the same data but in another react element but not sure why.

Reply:
check the latest invenio-rdm-records commits for usage.
Basically we use `AdminDetailsView` as a parent, and then we assign data prop to all its children in order to avoid making a separate request from another module